File: mock-facility-management/src/main/java/de/vkb/dojo/es/facilityManagement/kafka/config/TopicNames.java
package de.vkb.dojo.es.facilityManagement.kafka.config;

import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.boot.context.properties.ConstructorBinding;
import org.springframework.context.annotation.Configuration;

@ConfigurationProperties("spring.kafka.topics")
@Configuration
public class TopicNames {
    private String roomCommand;
    private String roomEventInternal;
    private String roomEventExternal;
    private String roomEventAndState;
    private String roomState;
    private String feedback;

    public String getRoomCommand() {
        return roomCommand;
    }

    public void setRoomCommand(String roomCommand) {
        this.roomCommand = roomCommand;
    }

    public String getRoomEventInternal() {
        return roomEventInternal;
    }

    public void setRoomEventInternal(String roomEventInternal) {
        this.roomEventInternal = roomEventInternal;
    }

    public String getRoomEventExternal() {
        return roomEventExternal;
    }

    public void setRoomEventExternal(String roomEventExternal) {
        this.roomEventExternal = roomEventExternal;
    }

    public String getRoomEventAndState() {
        return roomEventAndState;
    }

    public void setRoomEventAndState(String roomEventAndState) {
        this.roomEventAndState = roomEventAndState;
    }

    public String getRoomState() {
        return roomState;
    }

    public void setRoomState(String roomState) {
        this.roomState = roomState;
    }

    public String getFeedback() {
        return feedback;
    }

    public void setFeedback(String feedback) {
        this.feedback = feedback;
    }
}

File: todo-reservation-micronaut/src/main/kotlin/de/vkb/dojo/es/reservations/kafka/config/TopicNames.kt
package de.vkb.dojo.es.reservations.kafka.config

import io.micronaut.context.annotation.ConfigurationProperties

@ConfigurationProperties("kafka.topics")
interface TopicNames {
    val roomEvent: String
    val roomEventAndState: String
    val roomState: String
    val personEvent: String
    val personEventAndState: String
    val personState: String
    val reservationCommand: String
    val reservationEventInternal: String
    val reservationEventExternal: String
    val reservationEventAndState: String
    val reservationState: String
    val feedback: String
}
